lib/sycamore/sycamore/evaluation/evaluate.py

```diff
@@ -0,0 +1,139 @@
+from typing import Dict, List, Union
+from abc import ABC, abstractmethod
+
+import os
+from sycamore.data import Element, Document
+from sycamore.evaluation.pipeline import EvaluationPipeline
+from sycamore.evaluation import EvaluationDataPoint
+from sycamore.evaluation.metrics import document_retrieval_metrics, rouge_metrics
+from sycamore import Context
+
+
+def add_search_context_to_datapoint(datapoint: Document) -> List[Element]:
```

Comment:
Please add docstrings explaining what these functions do.

Reply:
Done, Added docustrings for all the classes and functions